packages/devtools_app/lib/src/ui/vm_flag_widgets.dart

```diff
@@ -14,24 +14,18 @@ import '../profiler/profile_granularity.dart';
 /// DropdownButton that controls the value of the 'profile_period' vm flag.
 ///
 /// This flag controls the rate at which the vm samples the CPU call stack.
-class ProfileGranularityDropdown extends StatefulWidget {
-  const ProfileGranularityDropdown(this.screenId);
+class ProfileGranularityDropdown extends StatelessWidget {
+  ProfileGranularityDropdown(this.screenId)
+      : profilerService = CpuProfilerService();
```

Comment:
creating a mutable profiler service in the constructor of a stateful widget seems like a bug. You don't want a new profiler service every time the widget rebuilds. Can you hang the service off the service manager, use a provider, or a stateful widget?

Reply:
Looking back through the original code, the methods on CpuProfilerService probably could have / should have been written as static. All they really are are helpers that interact with the vm service. Cleaned this up - PTAL